homeassistant/components/canary/camera.py

```diff
@@ -125,17 +128,36 @@ async def async_camera_image(
         self, width: int | None = None, height: int | None = None
     ) -> bytes | None:
         """Return a still image response from the camera."""
-        await self.hass.async_add_executor_job(self.renew_live_stream_session)
-        live_stream_url = await self.hass.async_add_executor_job(
-            getattr, self._live_stream_session, "live_stream_url"
-        )
-        return await ffmpeg.async_get_image(
-            self.hass,
-            live_stream_url,
-            extra_cmd=self._ffmpeg_arguments,
-            width=width,
-            height=height,
-        )
+        if self._image is None:
+            # if we still don't have an image, grab the live view
+            _LOGGER.debug("Grabbing a live view image from %s", self.name)
+            await self.hass.async_add_executor_job(self.renew_live_stream_session)
+
+            if self._live_stream_session is None:
+                return None
+
+            live_stream_url = self._live_stream_session.live_stream_url
+            if not live_stream_url:
+                return None
+
+            if self._image is None and live_stream_url:
```

Comment:
This check must be useless?
```suggestion
```

Reply:
I think you should accept this suggestion and add it back in the future PR.